catalog/spatial/kml/spatial-kml-transformer/src/main/java/org/codice/ddf/spatial/kml/transformer/KMLTransformerImpl.java

```diff
@@ -208,6 +228,94 @@ protected Placemark performDefaultTransformation(Metacard entry)
     return kmlPlacemark;
   }
 
+  private void setExtendedData(Placemark placemark, Metacard metacard) {
+    final ExtendedData extendedData = new ExtendedData();
+
+    final Set<AttributeDescriptor> attributeDescriptors =
+        metacard.getMetacardType().getAttributeDescriptors();
+
+    for (AttributeDescriptor attributeDescriptor : attributeDescriptors) {
+      final String attributeName = attributeDescriptor.getName();
+      final Attribute attribute = metacard.getAttribute(attributeName);
+      if (attribute != null) {
+        Serializable attributeValue = convertAttribute(attribute, attributeDescriptor);
+        if (attributeValue == null) {
+          LOGGER.debug("Attribute {} converted to null value.", attributeName);
+        } else {
+          final Data data = getData(attributeName, attributeValue.toString());
+          extendedData.addToData(data);
+        }
+      }
+    }
+    placemark.setExtendedData(extendedData);
+  }
+
+  private Data getData(String attributeAlias, String attributeValue) {
+    final Data data = new Data(attributeValue);
+    data.setName(attributeAlias);
+    return data;
+  }
+
+  private Serializable convertAttribute(Attribute attribute, AttributeDescriptor descriptor) {
+
+    if (descriptor.isMultiValued()) {
+      List<Serializable> values = new ArrayList<>();
+      for (Serializable value : attribute.getValues()) {
+        values.add(
+            convertValue(attribute.getName(), value, descriptor.getType().getAttributeFormat()));
+      }
+      String stringList = "";
+      for (int i = 0; i < values.size(); i++) {
+        if (i != 0) {
+          stringList += ", ";
+        }
+        stringList += values.get(i);
+      }
+      return stringList;
```

Comment:
I think you can `return StringUtils.join(values, ",");` and cut out this stringlist entirely